docs/how-tos/scale-applications-using-appgw-metrics.md

```diff
@@ -0,0 +1,107 @@
+# Scale your Applications using Application Gateway Metrics (Beta)
+
+As incoming traffic increases, it becomes crucial to scale up your applications based on the demand.
+
+In the following tutorial, we explain how you can use Application Gateway's `AvgRequestCountPerHealthyHost` metric to scale up your application. `AvgRequestCountPerHealthyHost` is measure of average request that are sent to a specific backend pool and backend http setting combination.
+
+We are going to use following two components:
+
+1. [`Azure K8S Metric Adapter`](https://github.com/Azure/azure-k8s-metrics-adapter) - We will using the metric adapter to expose Application Gateway metrics through the metric server.
+1. [`Horizontal Pod Autoscaler`](https://docs.microsoft.com/en-us/azure/aks/concepts-scale#horizontal-pod-autoscaler) - We will use HPA to use Application Gateway metrics and target a deployment for scaling.
+
+## Setting up Azure K8S Metric Adapter
+
+1. We will first create an Azure AAD service principal and assign it `Monitoring Reader` access over Application Gateway's resource group.
+    ```bash
+    applicationGatewayGroupName="<application-gateway-group-id>"
+    applicationGatewayGroupId=$(az group show -g $applicationGatewayGroupName -o tsv --query "id")
+    az ad sp create-for-rbac -n "azure-k8s-metric-adapter-sp" --role "Monitoring Reader" --scopes applicationGatewayGroupId
+    ```
+
+1. Now, We will deploy the [`Azure K8S Metric Adapter`](https://github.com/Azure/azure-k8s-metrics-adapter) using the AAD service principal created above.
+
+    ```bash
+    kubectl create namespace custom-metrics
+
+    # use values from service principle created above to create secret
+    kubectl create secret generic azure-k8s-metrics-adapter -n custom-metrics \
+        --from-literal=azure-tenant-id=<tenantid> \
+        --from-literal=azure-client-id=<clientid> \
+        --from-literal=azure-client-secret=<secret>
+
+    kubectl apply -f kubectl apply -f https://raw.githubusercontent.com/Azure/azure-k8s-metrics-adapter/master/deploy/adapter.yaml -n custom-metrics
+    ```
+
+1. We will create an `ExternalMetric` resource with name `appgw-request-count-metric`. This will instruct the metric adapter to expose `AvgRequestCountPerHealthyHost` metric for `myApplicationGateway` resource in `myResourceGroup` resource group. You can use the `filter` field to target a specific backend pool and backend http setting in the Application Gateway.
+
+    ```yaml
```

Comment:
I'd append here - ...copy paste this YAML content in `external-metric.yaml` and apply with `kubectl apply -f external-metric.yaml`